src/kOS.Safe/Compilation/Opcode.cs

```diff
@@ -1074,12 +1062,15 @@ public override void Execute(ICpu cpu)
                 Type t = value.GetType();
                 MethodInfo negateMe = t.GetMethod("op_UnaryNegation", BindingFlags.FlattenHierarchy |BindingFlags.Static | BindingFlags.Public); // C#'s alternate name for '-' operator
                 if (negateMe != null)
-                    result = negateMe.Invoke(null, new[]{value}); // value is an arg, not the 'this'.  (Method is static.)
+                {
+                    object result = negateMe.Invoke(null, new[]{value});
+                    scalarValue = ScalarValue.Create(result);
```

Comment:
This is wrong.  If you look at the comments above, one of the things that implements the unary negation operator are our vectors.  This line makes it impossible to return a vector from this negation.  Don't fix it - I'm just writing this here as a reminder to myself to fix it when I merge.